src/main/webapp/app/assessment/assessment-locks/assessment-locks.component.html

```diff
@@ -0,0 +1,192 @@
+                    <th jhiTranslate="artemisApp.assessment.dashboard.columns.submissionCount">Submission count</th>
+                    <th jhiTranslate="artemisApp.assessment.dashboard.columns.score">Score</th>
+                    <th jhiTranslate="artemisApp.assessment.dashboard.columns.action">Action</th>
+                </tr>
+            </thead>
+            <tbody>
+                <tr *ngFor="let fileUploadSubmission of fileUploadSubmissions; let i = index">
+                    <td>{{ i + 1 }}</td>
+                    <td>{{ fileUploadSubmission.id }}</td>
+                    <td>{{ getExerciseName(fileUploadSubmission.exerciseId) }}</td>
+                    <td>{{ fileUploadSubmission.submissionDate | date: 'MMM d, y HH:mm:ss' }}</td>
+                    <td>{{ fileUploadSubmission.participation.submissions.length }}</td>
+                    <td>
+                        <span *ngIf="fileUploadSubmission.result && fileUploadSubmission.result.score != null">{{ fileUploadSubmission.result.score }} %</span>
+                    </td>
+                    <td>
+                        <span *jhiHasAnyAuthority="['ROLE_ADMIN', 'ROLE_INSTRUCTOR', 'ROLE_TA']">
+                            <button
+                                *ngIf="fileUploadSubmission && fileUploadSubmission.result"
+                                [routerLink]="[
+                                    '/course-management',
+                                    courseId,
+                                    'file-upload-exercises',
+                                    fileUploadSubmission.exerciseId,
+                                    'submissions',
+                                    fileUploadSubmission.id,
+                                    'assessment'
+                                ]"
+                                class="btn btn-outline-secondary btn-sm mb-1"
+                            >
+                                <fa-icon [icon]="'folder-open'" [fixedWidth]="true"></fa-icon>&nbsp;{{ 'artemisApp.assessment.dashboard.actions.open' | translate }}
+                            </button>
+                        </span>
+                        <span>
+                            <button
+                                *ngIf="fileUploadSubmission && fileUploadSubmission.result && !fileUploadSubmission.result.completionDate"
+                                (click)="cancelFileUploadAssessment(fileUploadSubmission)"
+                                class="btn btn-outline-secondary btn-sm mb-1"
+                            >
+                                <fa-icon [icon]="'ban'" [fixedWidth]="true"></fa-icon>&nbsp;{{ 'artemisApp.assessment.dashboard.actions.cancel' | translate }}
+                            </button>
+                        </span>
+                    </td>
+                </tr>
+            </tbody>
+        </table>
+    </div>
+</div>
```

Comment:
Please extract this in a template or component as this is reused above.

Reply:
Refactored